File: samples/src/main/java/com/greenpepper/samples/application/bank/WithdrawType.java
package com.greenpepper.samples.application.bank;

/**
 * <p>WithdrawType class.</p>
 *
 * @author oaouattara
 * @version $Id: $Id
 */
public enum WithdrawType
{
	 ATM ("ATM"),
	 INTERAC("Interact"),
	 PERSONAL_CHECK ("Personal Check");

	private final String description ;

	private WithdrawType (String description)
	{
		this.description = description;
	}

	/** {@inheritDoc} */
	@Override
	public String toString()
	{
		return description;
	}
}

File: greenpepper-open/extensions-external/php/src/test/php/bank/WithdrawType.php
<?php

class WithdrawType 
{ 
  public static $ATM;
  public static $INTERAC;
  public static $PERSONAL_CHECK;
 
  private $description; 
  private $id;
  
  private static $counter = 0;
 
  private function WithdrawType($description) 
  { 
    $this->description = $description; 
	$this->id = WithdrawType::$counter;
	WithdrawType::$counter ++;
  } 
  
  public function toString() 
  { 
    return $this->description; 
  } 
  
  public function getId() {
	return $this->id;
  }
  
  public static function init() {  
	WithdrawType::$ATM = new WithdrawType("ATM");
	WithdrawType::$INTERAC = new WithdrawType("Interact");
	WithdrawType::$PERSONAL_CHECK = new WithdrawType("Personal Check"); 
  }
} 

WithdrawType::init();
